### DISCUSSION TITLE
I hope the zoom function becomes natural, too.

### DISCUSSION THREAD

**Discussion by aa53420** [linearmouse/linearmouse#333] [General]
This message was written through Google Translator.

Thank you for making such a wonderful app. My brain is no longer struggling.
But when I zoom in and out, my brain stumbles.
My wish is that zoom in & out will be more natural.

I hope you have a good day.

**Comment by lujjjh** [comment]
Could you please explain it further?

Do you mean zoom in and out by pressing <kbd>command</kbd><kbd>+</kbd> and <kbd>command</kbd><kbd>-</kbd>, or by some shortcuts / gestures on your mouse / trackpad?

And, how can we make it natural?

**Reply by aa53420** [reply]
Thank you so much for checking.

I usually zoom in && out.
Use the Cmmand + mouse wheel.

- Zoom In
`Command` `+` `Mouse wheel up`

- Zoom Out
`Command` `+` `Mouse wheel down`

It's different for everyone, but when I do Scroll reverse, I feel like this is natural.

**Reply by lujjjh** [reply]
I see. Thanks for your feedback!

You want to map command + mouse wheel up / mouse wheel down to zoom in / zoom out.

This feature is in my backlog and there are some problems remain unsolved. You may track it in #227.

**Reply by lujjjh** [reply]
This feature is now supported in [v0.7.7-beta.2](https://github.com/linearmouse/linearmouse/releases/tag/v0.7.7-beta.2).

**Reply by qns7** [reply]
Hey lujjjh: How do I achieve this? I want to map CMD++ to CMD+ScrollUp & CMD+- to CMD+ScrollDown (App-specific for 'Preview'). The latter works perfectly, but somehow the Hotkey-Recorder doens't recognize my +-input. I use smart-zoom instead for now (bound to CMD+ScrollUp), but it would be awesome if you reply to me. Thanks :)

PS: I have another similar issue with:
CTRL+ScrollUp -> MissionControl
CTRL+ScrollDown -> App-Exposé
Both worked for a few days, but now stopped working. 
Reinstallation and Restart both don't work wonders here. 
Best regards Q

**Reply by lujjjh** [reply]
Hey @qns7,

I haven't been able to reproduce the first issue. Mapping <kbd>cmd</kbd><kbd>=</kbd> or <kbd>cmd</kbd><kbd>shift</kbd><kbd>=</kbd> (which is <kbd>cmd</kbd><kbd>+</kbd> on on the US layout) works without any problems. However, you might want to try removing the button mappings and adjusting the Scrolling settings. Set the command modifier key to 'Zoom'. This will map <kbd>cmd</kbd><kbd>scroll up</kbd> and <kbd>cmd</kbd><kbd>scroll down</kbd> to <kbd>cmd</kbd><kbd>numpad+</kbd> and <kbd>cmd</kbd><kbd>numpad-</kbd>, respectively. Could you provide your keyboard layout so I can attempt to reproduce the issue?

Regarding the second issue, have you tried mapping them to other mouse buttons? It's possible that another mouse utility is also listening for <kbd>ctrl</kbd><kbd>scrollup</kbd> and <kbd>ctrl</kbd><kbd>scrolldown</kbd>. If the problem persists, please attach the configuration file and logs so I can see if there's any apparent issue.

**Reply by qns7** [reply]
Hey there,

thank you for your fast reply. I use the German/ISO layout. And it still
doesn’t work (cmd+). However using the zoom-function in 'scrolling'
works fine. It's a bit laggy though (fast scrolling doesn't quite work,
one has to go step-by-step), it's not a smooth scrolling experience. But
it’s totally fine for me, so thank you for telling me about this option. :)
But weirdly enough: If I manually map cmdscroll up and cmdscroll down to
cmdnumpad+ and cmdnumpad- within 'Buttons', it won’t work either.

I am not at home right now, but as soon as I am I will send you a log
file. On the other hand: I will try some things by manipulating the
linearmouse.json directly and see if that works in regard of cmd+ and
the scroll-stuff of issue [#2](https://github.com/linearmouse/linearmouse/pull/2). I also think about if maybe some other
utility interacts here, but I do not think so right now.

Best regards and till then
JK

EDIT: Attached the config-json-file.
[linearmouse.json](https://github.com/linearmouse/linearmouse/files/15464713/linearmouse.json)
I tried the ScrollUp/MissionControl-ScrollDown/AppExposé-thing again, but it just doesn't work. But I am comfortable using button 3 and 4 now, so don't worry.
Equally the ScrollUp/CMD+ still doesn't work - even if I changed it in the config manually to CMD+ by setting it to some other stuff beforehand. 
I just can't figure out why it behaves this way now, despite it working at the beginning of using linearmouse [at least the ScrollUp/MissionControl-...-thing worked fine and suddenly it stopped working].
And also that it works on your machine is kind of strange.
I can't tell if perhaps another app is interfering here and if I installed that villian exactly in-time when it stopped working - I don't know; it's most likely I guess.
But as I said, I'm happy now with the setup as it is after optimizing and setting up for days [among other things I use PDFgear now where zooming is a breeze].
So I just thank you for your effort, help and this awesome app linearmouse! <3
